packages/sentry/SentryErrorIntegration.tsx

```diff
@@ -0,0 +1,38 @@
+import { SentryError } from '.';
+import * as Sentry from '@sentry/nextjs';
+import {
+   BaseTransportOptions,
+   Client,
+   ClientOptions,
+   EventProcessor,
+   Hub,
+   Integration,
+} from '@sentry/types';
+
+export class SentryErrorIntegration implements Integration {
+   static id = 'SentryErrorIntegration';
+   name = 'SentryErrorIntegration';
+   setupOnce(
+      _addGlobalEventProcessor: (callback: EventProcessor) => void,
+      _getCurrentHub: () => Hub
+   ): void {}
+   processEvent?(
+      event: Sentry.Event,
+      hint: Sentry.EventHint,
+      client: Client<ClientOptions<BaseTransportOptions>>
+   ): Sentry.Event | PromiseLike<Sentry.Event | null> | null {
+      const exception = hint.originalException;
+
+      if (exception instanceof SentryError) {
+         const currentScope = Sentry.getCurrentHub().getScope();
+         const newScope = Sentry.Scope.clone(currentScope);
+
+         newScope.update(exception.sentryDetails);
+         newScope.applyToEvent(event, hint);
+      }
+
+      return event;
+   }
+
+   isDefaultInstance = true;
```

Comment:
I don't know what this property does, based on the [docs](https://docs.sentry.io/platforms/javascript/configuration/integrations/custom/?original_referrer=https%3A%2F%2Fwww.google.com%2F) and the [interface](https://github.com/getsentry/sentry-javascript/blob/master/packages/types/src/integration.ts)

Reply:
It's used to ensure we don't overwrite it with another version of the extension: https://github.com/getsentry/sentry-javascript/blob/ad4f2d1be6fb7ae4ca7f3e1da09bb8e0de2fde84/packages/core/src/integration.ts#L34-L38

It's a small safety net. I figure it couldn't hurt to set it.